packages/web3-utils/src/utils.js

```diff
@@ -233,7 +234,11 @@ var hexToNumber = function (value) {
         throw new Error('Given value "'+value+'" is not a valid hex string.');
     }
 
-    return toBN(value).toNumber();
+    const n = toBN(value);
+    if(n > Number.MAX_SAFE_INTEGER || n < Number.MIN_SAFE_INTEGER) {
+        return n.toString();
```

Comment:
This will be breaking change in case of over or underflows, and we can't do 5.x for this breaking change. 

Can you see if we can have same fix like https://github.com/web3/web3.js/issues/5432#issuecomment-1246483147 in our lib where we are using `hexToNumber`, so user can explicitly pass flag if he want string. 
```
var hexToNumber = function (value, numInStr = false) {
 // ..... 
    const n = toBN(value);
    if(numInStr){
        return n.toString();
    }
    else{
        return n.toNumber();
    }
};
```
or if above proposal is not perfect we can apply some other fix but should avoid breaking change 1.x.